addons/pos_razorpay/models/pos_payment_method.py

```diff
@@ -0,0 +1,122 @@
+                'p2pRequestId': '%s' % response.get('p2pRequestId')
+            }
+        default_error_msg = _('Razorpay payment request expected errorCode not found in the response')
+        error = response.get('errorMessage') or default_error_msg
+        return {'error': '%s' % error}
+
+    def razorpay_fetch_payment_status(self, data):
+        body = self._razorpay_get_payment_status_request_body()
+        body['origP2pRequestId'] = data.get('p2pRequestId')
+        response = self._call_razorpay(endpoint="status", payload=body)
+        if response.get('success') and not response.get('errorCode'):
+            _logger.info(response.get('message'))
+            payment_status = response.get('status')
+            if payment_status == 'AUTHORIZED' and response.get('messageCode') == 'P2P_DEVICE_TXN_DONE':
+                return {
+                    'status': response.get('status'),
+                    'authCode': response.get('authCode'),
+                    'cardLastFourDigit': response.get('cardLastFourDigit'),
+                    'externalRefNumber': response.get('externalRefNumber'),
+                    'txnId': response.get('txnId'),
+                    'paymentMode': response.get('paymentMode'),
+                    'paymentCardType': response.get('paymentCardType'),
+                    'paymentCardBrand': response.get('paymentCardBrand'),
+                    'acquirerCode': response.get('acquirerCode'),
+                    'settlementStatus': response.get('settlementStatus'),
+                    'createdTime': response.get('createdTime'),
+                    'userAgreement': response.get('userAgreement')
+                }
+            elif payment_status == 'FAILED' or response.get('messageCode') == 'P2P_DEVICE_CANCELED':
+                return {'error': "%s" % response.get('message', _('razorpay transaction failed'))}
+            else:
+                return {}
+        default_error_msg = _('Razorpay payment status request expected errorCode not found in the response')
+        error = response.get('errorMessage') or default_error_msg
+        return {'error': '%s' % error}
+
+    def razorpay_cancel_payment_request(self, data):
+        body = self._razorpay_get_payment_request_body()
+        body.pop('mode')
+        body['origP2pRequestId'] = data.get('p2pRequestId')
+        response = self._call_razorpay(endpoint="cancel", payload=body)
+        if response.get('success') and not response.get('errorCode'):
+            return {'error': "%s" % _('razorpay transaction canceled successfully')}
+        default_error_msg = _('Razorpay payment cancel request expected errorCode not found in the response')
+        error = response.get('errorMessage') or default_error_msg
+        return {'error': '%s' % error}
+
+    def _razorpay_get_payment_request_body(self):
+        additional_parameters = {
+            'pushTo': {
+                "deviceId": self.razorpay_tid + "|ezetap_android",
```

Comment:
```suggestion
                "deviceId": f"{self.razorpay_tid}|ezetap_android",
```
Just to stay on the safer side